### PR TITLE
redis: support for producing to dedicated listener redis instance

### PR DESCRIPTION
**Pull request by igorwwwwwwwwwwwwwwwwwwww**
refs travis-pro/team-teal#2064

### CODE REVIEW

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
@meatballhat yep, the plan is to revert this after the migration.

my plan is to move from a `redis` config parameter to a `listener_redis` one, to make it clear which instance we are talking about in a global context, similar to `logs_database`. but we can discuss the details of that in the dedicated PR for that. :)

File: lib/travis/listener.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ def config
 
   module Listener
     class Config < Travis::Config
-      define  redis:   { url: 'redis://localhost:6379', namespace: 'sidekiq', network_timeout: 5 },
+      define  redis:   { url: ENV['LISTENER_REDIS_URL'] || 'redis://localhost:6379', namespace: 'sidekiq', network_timeout: 5 },
```
Comment (meatballhat) [inline lib/travis/listener.rb]: With an addon allocated to the listener app (which is my preference, which iiuc is not currently the case), I would expect `REDIS_URL` to be the default.  Is there a reason why we need the env var within the listener app to be `LISTENER_REDIS_URL`?

Comment (igorwwwwwwwwwwwwwwwwwwww) [inline lib/travis/listener.rb]: My preference for `LISTENER_REDIS_URL` is to have consistent naming across all apps, and not have them be context specific. I like the idea of being able to look at the code and instantly tell which redis instance is being referred to.

I am open to changing it to `REDIS_URL` though. Let me know what you think.

Comment (igorwwwwwwwwwwwwwwwwwwww) [inline lib/travis/listener.rb]: In fact, I would even be in favour of renaming the config key to `listener_redis`.

Comment (meatballhat) [inline lib/travis/listener.rb]: I'm +0 to using the same env var across apps.  For, me the fact that a common env var like `REDIS_URL` or `DATABASE_URL` has a prefix tells me that it "belongs" to another app.  The specific case of `LOGS_DATABASE_URL` on the travis-logs apps is a leftover from a transitional time that I would like to change to `DATABASE_URL`.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
@meatballhat It's sort of not even clear if the db should be owned by listener or gatekeeper. Considering it is gatekeeper's _incoming_ queue, maybe it would be better to have gatekeeper own it.

Since in this case the database is a shared piece of infrastructure _between_ services as opposed to one owned by a particular service (and isolated to that service) I am still in favour of naming it explicitly.

**Comment by meatballhat** [comment]
@igorwwwwwwwwwwwwwwwwwwww totally understood regarding shared resource.  In that case, should the redis instance be named something that indicates how it is used rather than to which app it "belongs"?  Something like `GITHUB_EVENTS_REDIS_URL`?

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
I suppose we could go with `BUILD_REQUESTS`, but I think it would be conceptually simpler to allocate the name to the incoming app. So I would be in favour of `REDIS_GATEKEEPER_URL`.

wdyt?

**Comment by meatballhat** [comment]
I think that this is a highly subjective topic 😁 Naming is hard.  `REDIS_GATEKEEPER_URL` SGTM.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
🙇‍♀️ thank you!

**Comment by meatballhat** [comment]
🙀 oh no we talked about gatekeeper in a public PR

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
😵😂

**Pull request merged**